pkg/controller/master/virtualmachine/register.go

```diff
@@ -30,28 +43,42 @@ func Register(ctx context.Context, management *config.Management, options config
 		vmiCache       = vmiClient.Cache()
 		vmBackupClient = management.HarvesterFactory.Harvesterhci().V1beta1().VirtualMachineBackup()
 		vmBackupCache  = vmBackupClient.Cache()
+		vmimCache      = management.VirtFactory.Kubevirt().V1().VirtualMachineInstanceMigration().Cache()
 		snapshotClient = management.SnapshotFactory.Snapshot().V1beta1().VolumeSnapshot()
 		snapshotCache  = snapshotClient.Cache()
 	)
 
+	copyConfig := rest.CopyConfig(management.RestConfig)
+	copyConfig.GroupVersion = &kubevirtSubResouceGroupVersion
+	copyConfig.APIPath = "/apis"
+	copyConfig.NegotiatedSerializer = scheme.Codecs.WithoutConversion()
+	virtSubresourceClient, err := rest.RESTClientFor(copyConfig)
```

Comment:
the virtSubresourceClient(rest client) is only being used to stop the VM when there is not enough capacity in the resource quote.
This can easily be done with the vmClient already available in the handler. For example: https://github.com/harvester/harvester/blob/master/pkg/controller/master/nodedrain/nodedrain_controller.go#L134-L139